docs/UserGuide.md

```diff
@@ -52,198 +53,292 @@ Tips are helpful suggestions that may improve your experience using TutorAid.
+
+
+### 4.3: Adding a student to a lesson
+Now that you have set up the lesson _History 1_, let us add one of your existing students who wishes to enrol in the class, Roy Balakrishnan. You can do this in TutorAid by following the steps below:
+
+1. Key in `add -sl s/6 l/2` into the Command Box and press ENTER.
+
+    > :bulb: The numbers `6` and `2` in the command above refer to the indexes of Roy and _History 1_ in the Student and Lesson Panels respectively. For more information, you can refer to the detailed documentation for [adding a student to a lesson](#adding-students-to-lessons-add--sl).
+
+1. As shown in the image below, you should now be able to see that in the Student Panel, the lesson _History 1_ has been added to the list of lessons taken by Roy. Similarly, in the Lesson Panel, Roy has been added to the list of students attending _History 1_. 
+
+   ![](images/addToLesson.png)
+
+### 4.4: Adding a progress note for a student
 {:.no_toc}
 Now, imagine you have ended your 'Maths 1' lesson. You realised that one of your students, Alex Yeoh, seems to struggle with the topic 'Vectors', and you wish to take 
 note of it so that you can provide him extra practice. To do so, you can easily add a progress note for Alex on TutorAid by following the steps below: 
-1. Key in `find -s Alex Yeoh` into the command box. This would result in only Alex's details being displayed in the student panel.
-2. Key in `add -p 1 Weak in Vectors and requires extra practice` into the command box. Alex's details will then be updated to display the progress note you just added. 
 
-Alternatively, you can also do this:
-1. Key in `list` in the command box to see list of all your students in the student panel.
-2. Locate `Alex Yeoh` in the student panel and take note of his index. In this case, it would be `1`.
-3. Key in `add -p 1 Weak in Vectors and requires extra practice` into the command box. Alex's details will then be updated to display the progress note you just added.
+1. Key in `add -p 1 Requires extra practice for Vectors` into the Command Box and press ENTER. 
+
+    > :bulb: The `1` in the command above refers to the index number of Alex Yeoh in the Student Panel. For more information, you can refer to the detailed documentation for the [Add a Progress to a Student](#adding-progress-for-a-student-add--p) command.
+   
+1. Notice that Alex's details reflected in the Student Panel have updated to display the progress note you just added, as seen here:
+
+    ![](images/addProgress.png)
+   
+   You can then key in the `list -a` into the Command Box and hit ENTER to return back to your original view.
 
-Now, the next time you open TutorAid, you will be able to see Alex's progress note at a glance! 
+   > :bulb: For more information on changing the viewing modes in TutorAid, you can refer to the detailed documentation for [List](#listing-all-students-and-lessons-list) commands.
```

Comment:
I repeated this wherever the `list -a` command is used because I thought we can't really assume that they are going through every part of the tutorial, so if they skip an earlier part then they won't have an idea of where this command came from. But I understand that it is a little repetitive as well, I will see if I can find a good place to just add this comment once